core-common/src/main/java/org/glassfish/jersey/logging/LoggingInterceptor.java

```diff
@@ -312,4 +332,24 @@ public void write(byte[] ba, int off, int len) throws IOException {
         }
     }
 
+    private static final class RedactHeaderPredicate implements Predicate<String> {
+        private final Set<String> headersToRedact;
+
+        RedactHeaderPredicate(Collection<String> headersToRedact) {
+            this.headersToRedact = headersToRedact.stream()
+                    .filter(Objects::nonNull)
+                    .filter(Predicates.not(String::isEmpty))
+                    .map(RedactHeaderPredicate::normalize)
+                    .collect(Collectors.toSet());
+        }
+
+        @Override
+        public boolean test(String header) {
+            return headersToRedact.contains(normalize(header));
+        }
+
+        private static String normalize(String input) {
+            return input.trim().toLowerCase(Locale.ROOT);
```

Comment:
Just crossed my mind that maybe we should do a null check here. Not sure if it's possible for a null header ever to reach this point but could do it just to be on the safe side.
Thoughts, @jansupol @senivam?

Reply:
I think it's OK. This code ``final String header = headerEntry.getKey();`` is NPE safe and this code
 ```
 this.headersToRedact = headersToRedact.stream()
                    .filter(Objects::nonNull)
                    .filter(Predicates.not(String::isEmpty))
                    .map(RedactHeaderPredicate::normalize)
                    .collect(Collectors.toSet());
```
 is NPE safe. Nothing else calls that predicate.